src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -317,18 +319,21 @@ public void init(AdaptorContext context) throws Exception {
    * @param sql the SQL query; the query is prepared, but not executed
    * @param columnConfig the configuration property for the column names
    * @param columnNames the column names to verify
+   * @return a Map of columnName to Integer java.sql.Types
```

Comment:
Look at how UniqueKey maps multiple SQL types to ColumnType.  Moving that logic here seems very out-of-place.  But that does not really matter.  When I make a UniqueKey.Builder, it will likely take a ResultSet (or ResultSetMetaData), rather than this map.